.github/workflows/build_and_test.yml

```diff
@@ -99,6 +99,14 @@ jobs:
           path: |
             rskj-core/build/reports/
 
+      - name: Upload HTML Test Report
```

Comment:
wondering if we should always upload results or only if the tests are failing?

Reply:
To me, it makes sense only when the tests failed

Reply:
Addressed